app/models/post.rb

```diff
@@ -2,4 +2,13 @@ class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
   has_many :comments
   has_many :likes
+
+  def update_posts_counter
+    author.posts_counter = author.posts.count
+    author.save
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
```

Comment:
- [ Optional ] Kindly I recommend you to use  [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to database, then make the `update_posts_counter` method private just like the below sample **after_save** is invoked every time a record is saved to the database, whether it is a new record or an existing record that is updated.

```
def recent_comments
    comments.order(created_at: :desc).limit(5)
  end

after_save :update_posts_counter

private
  def update_posts_counter
     author.posts_counter = author.posts.count
    author.save
  end

```